thor_devkit/transaction.py

```diff
@@ -303,23 +423,50 @@ def encode(self):
             temp.update({
                 'signature': self.signature
             })
-            return ComplexCodec(SignedTxWrapper).encode(temp)
+            if self.get_type() == TransactionType.DYNAMIC_FEE:
+                return ComplexCodec(EIP1559SignedTxWrapper).encode(temp)
+            else:
+                return ComplexCodec(LegacySignedTxWrapper).encode(temp)
         else:
-            return ComplexCodec(UnsignedTxWrapper).encode(temp)
+            if self.get_type() == TransactionType.DYNAMIC_FEE:
+                return ComplexCodec(EIP1559UnsignedTxWrapper).encode(temp)
+            else:
+                return ComplexCodec(LegacyUnsignedTxWrapper).encode(temp)
 
     @staticmethod
     def decode(raw: bytes, unsigned: bool):
         ''' Return a Transaction type instance '''
         body = None
         sig = None
 
-        if unsigned:
-            body = ComplexCodec(UnsignedTxWrapper).decode(raw)
-        else:
-            decoded = ComplexCodec(SignedTxWrapper).decode(raw)
-            sig = decoded['signature']  # bytes
-            del decoded['signature']
-            body = decoded
+        # First try to decode as dynamic fee transaction
+        try:
+            if unsigned:
+                body = ComplexCodec(EIP1559UnsignedTxWrapper).decode(raw)
+            else:
+                decoded = ComplexCodec(EIP1559SignedTxWrapper).decode(raw)
+                sig = decoded['signature']  # bytes
+                del decoded['signature']
+                body = decoded
+            body['type'] = TransactionType.DYNAMIC_FEE.value
+        except:
+            # If that fails, try legacy transaction
+            if unsigned:
+                body = ComplexCodec(LegacyUnsignedTxWrapper).decode(raw)
+            else:
+                decoded = ComplexCodec(LegacySignedTxWrapper).decode(raw)
+                sig = decoded['signature']  # bytes
+                del decoded['signature']
+                body = decoded
+            body['type'] = TransactionType.NORMAL.value
+
+        # Convert numeric values to hex strings
+        if 'nonce' in body:
+            body['nonce'] = hex(body['nonce'])
+        if 'maxFeePerGas' in body:
+            body['maxFeePerGas'] = hex(body['maxFeePerGas'])
```

Comment:
Do we have to present all fees as hex ? And why ?
Python has built-in big numbers to represent anything so we don't have to use hex notation when not needed.